pkg/vault/test/foundry/BufferVaultPrimitive.t.sol

```diff
@@ -788,23 +798,33 @@ contract BufferVaultPrimitiveTest is BaseVaultTest {
         (uint256 daiIdx, uint256 waDaiIdx, IERC20[] memory tokens) = _getTokenArrayAndIndexesOfWaDaiBuffer();
         BaseVaultTest.Balances memory balancesAfter = getBalances(lp, tokens);
 
-        // Check wrap results. For wrap exact out, when the buffer has enough liquidity to fulfill the operation,
-        // amount in increases by conversion factor.
-        uint256 convertFactorIn = withBufferLiquidity && kind == SwapKind.EXACT_OUT ? vaultConvertFactor : 0;
-        assertEq(amountIn, _wrapAmount + convertFactorIn, "AmountIn (underlying deposited) is wrong");
-        assertEq(
+        // Check wrap results.
+        // For wrap exact out, when the buffer has enough liquidity to fulfill the operation, amountIn increases by
+        // conversion factor. Depending on the token rate, the value may change a bit, but it's important that amountIn
+        // is bigger than _wrapAmount to make sure the buffer is not drained.
+        uint256 convertFactorExactOut = withBufferLiquidity && kind == SwapKind.EXACT_OUT ? vaultConvertFactor : 1;
+        assertApproxEqAbs(
+            amountIn,
+            _wrapAmount + convertFactorExactOut,
+            convertFactorExactOut,
+            "AmountIn (underlying deposited) is wrong"
+        );
+        assertApproxEqAbs(
             balancesAfter.lpTokens[daiIdx],
-            balancesBefore.lpTokens[daiIdx] - _wrapAmount - convertFactorIn,
+            balancesBefore.lpTokens[daiIdx] - _wrapAmount - convertFactorExactOut,
+            convertFactorExactOut,
```

Comment:
This one will probably change in upcoming PRs right?

Reply:
Yes, exactly! #967 changes it.